crates/solver/src/settlement_rater.rs

```diff
@@ -244,18 +249,198 @@ impl SettlementRating for SettlementRater {
+            Ok(objective.clone()
+                - probability_fail / probability_success * (self.score_cap.clone() + cost_fail))
+        } else if payoff_obj_minus_cap < BigRational::zero() && payoff_cap <= BigRational::zero() {
+            Ok(probability_success / probability_fail * self.score_cap.clone() - cost_fail)
+        } else {
+            Err(anyhow!("Invalid bid"))
+        };
+        tracing::trace!(?optimal_bid, "Optimal bid");
+        if let Ok(optimal_bid) = optimal_bid.as_ref() {
+            ensure!(
+                optimal_bid <= &objective,
+                "Optimal score higher than objective value"
+            );
+        }
+        optimal_bid
+    }
+
+    fn payoff(
+        &self,
+        score_reference: BigRational,
+        objective: BigRational,
+        probability_success: BigRational,
+        cost_fail: BigRational,
+    ) -> BigRational {
+        tracing::trace!(
+            ?score_reference,
+            ?objective,
+            ?probability_success,
+            ?cost_fail,
+            "Computing payoff"
+        );
+        let probability_fail = BigRational::one() - probability_success.clone();
+        let payoff_success = min(objective - score_reference.clone(), self.score_cap.clone());
+        tracing::trace!(?payoff_success, "Payoff success");
+        let payoff_fail = -min(score_reference, self.score_cap.clone()) - cost_fail;
+        tracing::trace!(?payoff_fail, "Payoff fail");
+        let payoff_expectation =
+            probability_success * payoff_success + probability_fail * payoff_fail;
+        tracing::trace!(?payoff_expectation, "Payoff expectation");
+        payoff_expectation
+    }
+}
+
+mod tests {
+    use {num::BigRational, primitive_types::U256};
+
+    #[allow(dead_code)]
+    fn calculate_score(objective_value: BigRational, success_probability: f64) -> U256 {
+        let score_cap = BigRational::from_float(1e17).unwrap();
+        let score_calculator = super::ScoreCalculator::new(score_cap);
+        score_calculator
+            .compute_score(&objective_value, success_probability)
+            .unwrap()
+            .score()
+    }
+
+    #[test]
+    fn compute_score_with_success_probability_test() {
+        let objective_value = num::BigRational::from_float(251547381429604400.).unwrap();
+        let success_probability = 0.9202405649482063;
+        let score = calculate_score(objective_value, success_probability).to_f64_lossy();
+        assert_eq!(score, 250680657682686317.);
+    }
+
+    #[test]
+    fn compute_score_with_success_probability_test2() {
+        let objective_value = num::BigRational::from_float(1e16).unwrap();
+        let success_probability = 0.9;
+        let score = calculate_score(objective_value, success_probability).to_f64_lossy();
+        assert_eq!(score, 9e15);
+    }
+
+    #[test]
+    fn compute_score_with_success_probability_test3() {
```

Comment:
Let's take this case for instance. Here the objective value is the same as the cap (1e17), therefore the cap shouldn't matter. According to the doc, I'd expect the bid to be

```
probability_success * objective
```

which would be 2/3e17 not 0.95e17

What am I missing?

Reply:
The cap is `1e16` and this test falls in category 2 (large objective, large success probability).

Reply:
Updated cap to `1e16`, my bad